python/lsst/ts/observatory/control/auxtel/atcalsys.py

```diff
@@ -0,0 +1,417 @@
+        """Perform an electrometer scan for the specified duration.
+
+        Parameters
+        ----------
+        duration : `float`
+            Total duration of scan.
+
+        Returns
+        -------
+        RunTimeError
+
+        """
+
+        self.rem.electrometer_1.evt_largeFileObjectAvailable.flush()
+
+        try:
+            await self.electrometer.cmd_startScanDt.set_start(
+                scanDuration=duration, timeout=duration + self.long_timeout
+            )
+        except salobj.AckTimeoutError:
+            self.log.exception("Timed out waiting for the command ack. Continuing.")
+
+        # Make sure that a new lfo was created
+        lfo = await self.electrometer.evt_largeFileObjectAvailable.next(
+            timeout=self.long_timeout, flush=False
+        )
+        return lfo
+
+    async def take_fiber_spectrum(
+        self,
+        delay: float,
+        integration_time: float,
+    ) -> salobj.type_hints.BaseMsgType:
+        """Wait, then start an acquisition with the fiber spectrograph.
+
+        By default, this method will wait for `delay` seconds then start
+        an acquisition with the fiber spectrograph. Optionally the user may
+        provide a coroutine that will be awaited before the delay starts.
+
+        Parameters
+        ----------
+        delay : `float`
+            Seconds to wait before starting fiber spectrograph acquisition.
+        integration_time : `float`
+            Integration time for the fiber spectrum (seconds).
+
+        Returns
+        -------
+        RuntimeError : If exposure or return of lfo doesn't occur.
+        """
+        self.fiberspectrograph.evt_largeFileObjectAvailable.flush()
+
+        try:
+            await self.fiberspectrograph.cmd_expose.set_start(
+                integrationTime=integration_time,
+                timeout=integration_time + self.long_timeout,
+            )
+        except salobj.AckTimeoutError:
+            self.log.exception("Timed out waiting for the command ack. Continuing.")
+
+        lfo = await self.fiberspectrograph.evt_largeFileObjectAvailable.next(
+            timeout=self.long_timeout, flush=False
+        )
+        return lfo
+
+    async def perform_flat(self, exptime_dict: dict[str, float]) -> None:
+        # check dictionary
+        # Get the image number
+        # Somehow need to make sure camera is setup
+        if self.atcamera is not None:
+            asyncio.gather(
+                self.electrometer_scan(duration=exptime_dict["electrometer"]),
+                self.take_fiber_spectrum(
+                    delay=self.delay, integration_time=exptime_dict["fiberspectrograph"]
+                ),
+                self.atcamera.take_flats(exptime=exptime_dict["camera"], nflats=1),
+            )
+
+    def _get_monochromator_setup(
+        self,
+        filter: typing.Union[int, str, None],
+        calib_type: typing.Union[int, str, None],
```

Comment:
I think `calib_type` should probably be an enumeration. Having it as an int, str or None is not a good idea. 

Please, create an enumeration called `CalibrationType` and use it instead.

You can see some examples of enumerations [here](https://github.com/lsst-ts/ts_observatory_control/blob/develop/python/lsst/ts/observatory/control/utils/enums.py). Which is probably a good place to add this enumeration as well.

Reply:
I'm actually going to remove `_get_monochromator_setup` as we can just call these things from a configuration file as previously suggested. I will add in an enumeration for `CalibrationType` for getting optimized exposure times and it will be useful for the Simonyi Telescope